lib/mix/lib/mix/task.compiler.ex

```diff
@@ -162,4 +162,33 @@ defmodule Mix.Task.Compiler do
         {:noop, []}
     end
   end
+
+  @doc """
+  Reenables given compilers so they can be executed again down the stack.
+
+  If an umbrella project reenables compilers, they are re-enabled for all
+  child projects.
+
+  Default is `[]`, for none compilers to be reenabled.
+  This task always re-enables `"compiler.all"`.
+  """
+  @spec reenable([{:compilers, compilers}]) :: :ok when compilers: :all | [Mix.Task.task_name()]
+  def reenable(opts \\ []) do
+    if not Keyword.keyword?(opts) do
+      IO.warn(
+        "Mix.Task.Compiler.reenable/1 expects a keyword list " <>
+          "as an argument with compilers: [compiler()], " <>
+          "reenabling no compiler"
+      )
+    else
+      compilers =
+        case Keyword.get(opts, :compilers, []) do
+          :all -> Mix.Tasks.Compile.compilers()
+          list when is_list(list) -> list
+        end
+
+      Enum.each(["loadpaths", "compile", "compile.all"], &Mix.Task.reenable(&1))
```

Comment:
I am unsure if we should enable loadpaths. Thoughts? Why not `deps.loadpaths`as well? That's one of the slippery slopes that worry me. :)

Reply:
> I am unsure if we should enable loadpaths.

We should not. It is a leftover from the code when I thought the enabling should be mirrored by a roll-back, but it should not in the latest approach. I cannot think of any change resulting in a need to reload paths _from within same external task execution_, even if it calls `reenable/1`.